olp-cpp-sdk-dataservice-read/src/Common.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <olp/core/client/CancellationToken.h>
+#include <olp/dataservice/read/FetchOptions.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+/*
+ * @brief Common function to perform task separation for CacheWithUpdate fetch
+ * options.
+ * When user specifies CacheWithUpdate request, we start 2 tasks:
+ * - perform cache lookup and return user the result
+ * - perform online request and update cache
+ * @param request Request provided by the user.
+ * @param add_task_function Function used to setup a task, consumes the request.
+ * @param callback Operation callback specified by the user.
+ * @return CancellationToken used to cancel the operation.
+ */
+template <typename Request, typename Function, typename Callback>
+inline client::CancellationToken Execute(Request&& request,
```

Comment:
Name seems to generic, maybe something like `CacheWithUpdateOrExecute`

Reply:
Maybe just `FetchWithUpdate` or `CacheWithUpdate`

Reply:
Renamed it to AddTasks, add_task_function is now the first argument.